R/fd_chull_intersect.R

```diff
@@ -0,0 +1,46 @@
+# A wrapper around geometry::intersectn() to properly handle errors and specific
+# cases
+fd_chull_intersect <- function(traits1, traits2) {
+
+  traits1 <- traits1[!duplicated(traits1),, drop = FALSE]
+  traits2 <- traits2[!duplicated(traits2),, drop = FALSE]
+
+  if (ncol(traits1) == 1L & ncol(traits2) == 1L) {
+
+    r1 <- range(traits1)
+    r2 <- range(traits2)
+
+    # Range of the overlap
+    r3 <- c(max(r1[1], r2[1]),
+            min(r1[2], r2[2]))
+
+    r_overlap <- diff(r3)
```

Comment:
```suggestion
    r_overlap <- r3[2] - r3[1]
```

Better performance:

|expr               |  min|   lq|      mean| median|   uq|     max| neval|
|:------------------|----:|----:|---------:|------:|----:|-------:|-----:|
|{     1 - 2 }      |   84|  127|  161.1078|    137|  155| 4997253| 1e+06|
|{     diff(1, 2) } | 2214| 2407| 2760.8737|   2509| 2753| 9890056| 1e+06|